crates/rome_analyze/src/signals.rs

```diff
@@ -70,8 +74,12 @@ where
         Some(AnalyzerDiagnostic::from_error(error))
     }
 
-    fn action(&self) -> Option<AnalyzerAction<L>> {
-        (self.action)()
+    fn actions(&self) -> AnalyzerActionIter<L> {
+        if let Some(action) = (self.action)() {
+            AnalyzerActionIter::new(vec![action])
+        } else {
+            AnalyzerActionIter::new(vec![])
+        }
```

Comment:
No need to use a `vec` here
```suggestion
        if let Some(action) = (self.action)() {
            AnalyzerActionIter::new([action])
        } else {
            AnalyzerActionIter::new([])
        }
```